e2e/sanity_test.go

```diff
@@ -74,6 +75,7 @@ func testSanity() {
 	}
 
 	if os.Getenv("SANITY_TEST_WITH_THIN_DEVICECLASS") == "true" {
+		// This test does not run in the legacy mode.
```

Comment:
I'm not sure but is this test _need not_ to pass in the legacy mode?
If you mean that we do not run this test with the legacy mode in CI, it's just a limitation due to the bandwidth of CI runners. I think we should be able to run it in our local environment if we'd like to do so.